### ISSUE TITLE
iFrame support

### ISSUE THREAD

**Issue by argyleink** [GoogleChromeLabs/ProjectVisBug#265]
issue is similar to #186

**Comment by argyleink** [comment]
closing this, i dont think visbug could or should have visibility into actions taken on another domain in an iframe. users of visbug can open those iframes into their own tabs and work from there. 

open to ideas though!

**Comment by Cybamen** [comment]
Would like to have the support for iframes.

**Comment by argyleink** [comment]
can you tell me more about your use case?
did you try opening them into new tabs and debugging them there?

**Comment by Cybamen** [comment]
Yeah of course, the product is using a microservice architecture. Thats why the software uses a navigation system outside of the microservices. And the easiest way to bring two or more microservices together is via link or iframe. Its a business software.

**Comment by argyleink** [comment]
hm, yes, meaningful use case! thanks for sharing it in this thread 👍
i'll keep thinking about how to support iframes, perhaps we'll reopen this when we have a solid plan on how to do it?

**Comment by Cybamen** [comment]
Thanks

**Comment by hayes** [comment]
Hey, I have a use case that could really benefit from iframes.  Not supporting iframes from other domains makes sense, but I am curious if same-domain iframes could be supported.  

The use case is storybook.  storybook renders each story in an iframe.  Our team is looking to improve the review process for our designers, and VizBug seems to be an ideal tool for this, but opening each story in a new window is a friction point.  Totally understand if this is out of scope, but figured I'd leave a comment just in case.

**Comment by argyleink** [comment]
> Hey, I have a use case that could really benefit from iframe support. Not supporting iframes from other domains makes sense, but I am curious if same-domain iframes could be supported.

I appreciate the friendly ping 🙂  
same domain iframes could be possible yeah! it looks like (i've only looked at a few events) some of the essentials should work. I'm going to reopen this, a prototype could get made to see how it works and how much effort it'll be to get all the tools working inside of there. I imagine there's some critical changes that need made to not assume it's all on the window document, but we'll see 🤓

**Comment by miguel-sf** [comment]
> closing this, i dont think visbug could or should have visibility into actions taken on another domain in an iframe. users of visbug can open those iframes into their own tabs and work from there.
> 
> open to ideas though!

I have an app that is rendered inside another app, needs the top auth to even be rendered. How can I use visbug in this case if I can't open the iFrame by itself as it wont pass auth

**Comment by muchisx** [comment]
> closing this, i dont think visbug could or should have visibility into actions taken on another domain in an iframe. users of visbug can open those iframes into their own tabs and work from there.
> 
> open to ideas though!

I have an app that is rendered inside another app, needs the top auth to even be rendered. How can I use visbug in this case if I can't open the iFrame by itself as it wont pass auth

**Comment by argyleink** [comment]
not sure.. currently a limitation of extensions and where they can be invoked plus iframes being a boundary..

if you really needed it, you could try and inject the visbug web component into the iframe with a script, but I dont have anything prepared for you to do that.. the good news is that it's just a web component and all the extension does is facilitate showing/hiding the component and telling it where the img tutorial resources are. you could def put it in there if you really wanted to, just without much convenience
